### PR TITLE
show password checkbox gets enabled only on clicking on the Checkbox #3267

### LINKED ISSUE #3267: `show password` checkbox gets enabled on clicking outside the label
### What is the expected behaviour?

On the login screen `show password` checkbox should not be enabled when clicked outside the label.

### What is the current behaviour?

The `show password` gets enabled when clicked outside the label
 
### How to reproduce the issue?

1. Go to the login screen of ToolJet
2. Click anywhere on the right side of the `show password`

### Screenshots or Screencast
show-password

### PR DESCRIPTION
**Pull request by Syed-Ansar**
show password checkbox gets enabled only on clicking on the Checkbox 

- fixes #3267  

https://user-images.githubusercontent.com/82027712/174958121-d68eb3d6-4642-409c-bb0c-ec3b4a9f79d3.mp4

### CODE REVIEW

**Comment by withshubh** [comment]
Please do not merge unless approved.

File: frontend/src/_styles/theme.scss [hunk 1]
```diff
@@ -1877,6 +1877,10 @@ button {
   border-color: none;
 }
 
+.form-check-input{
```
Comment (shah21) [inline frontend/src/_styles/theme.scss:1880]: Please add the cursor to the label also

File: frontend/src/LoginPage/LoginPage.jsx [hunk 2]
```diff
@@ -199,7 +199,7 @@ class LoginPage extends React.Component {
                         onChange={this.handleOnCheck}
                         data-cy="checkbox-input"
                       />
-                      <label className="form-check-label" htmlFor="check-input" data-cy="show-password-label">
+                      <label className="form-check-label" htmlFor="check-Input" data-cy="show-password-label">
```
Comment (shah21) [inline frontend/src/LoginPage/LoginPage.jsx:202]: Please change the `htmlFor` value back to "check-input" while the id of checkbox input is "check-input". If you change the `htmlFor` value, we will not be able to `enable or disable` the checkbox by clicking the label. so, please use CSS properties like 
```
.show-password-label {
 width: fit-content;
}
```

**Review by shah21 — COMMENTED** [review]
Hey, @Syed-Ansar 
I have left some comments, please take a look

File: frontend/src/LoginPage/LoginPage.jsx (hunk 2 above)
Comment (Syed-Ansar) [reply frontend/src/LoginPage/LoginPage.jsx:202]: Hey @shah21 the issue I fixed through this PR is that we should not be able to enable or disable checkout by clicking on label show password #3267

**Comment by Syed-Ansar** [comment]
> Hey, @Syed-Ansar I have left some comments, please take a look

Hey @shah21 can you look after the issue #3267

**Comment by shah21** [comment]
> > Hey, @Syed-Ansar I have left some comments, please take a look
> 
> Hey @shah21 can you look after the issue #3267

Yes, I saw the issue, and I found this
`On the login screen show password checkbox should not be enabled when clicked outside the label.`
So what we need to do is prevent enable or disable by clicking outside the label and input

**Comment by Syed-Ansar** [comment]
> > > Hey, @Syed-Ansar I have left some comments, please take a look
> > 
> > 
> > Hey @shah21 can you look after the issue #3267
> 
> Yes, I saw the issue, and I found this `On the login screen show password checkbox should not be enabled when clicked outside the label.` So what we need to do is prevent enable or disable by clicking outside the label and input

Oh I thought it should not be able to enable or disable checkout box. I will resolve the suggested changes.

File: frontend/src/_styles/theme.scss [hunk 3]
```diff
@@ -1877,6 +1877,18 @@ button {
...
 
+.form-check-input {
+  cursor: pointer;
+}
+
+.show-password-label {
+  width: fit-content;
+ }
+
+ .form-checks {
```
Comment (shah21) [inline frontend/src/_styles/theme.scss:1888]: please use appropriate names for classes
like `.show-password-field` for` .show-password-label`
and `.show-password-label` for ` form-checks` 

And can we put the .show-password-label class inside .show-password-field ?
like 
```
.show-password-field {
  .show-password-label {
  }
 }
```

**Review by shah21 — COMMENTED** [review]
A small change @Syed-Ansar 
Also, please resolve the lint errors

File: frontend/src/_styles/theme.scss (hunk 3 above)
Comment (Syed-Ansar) [reply frontend/src/_styles/theme.scss:1888]: Hey @shah21 I have created another PR #3428  please review If you satisfy with that I will close this PR.

**Comment by withshubh** [comment]
Fixed in #3428

**Pull request closed without merging**